PEP 662: Editable installs via virtual wheels (#1977)
python/peps: pep-9999.rst

PEP: 9999
Title: Editable installs
Author: Bernát Gábor <gaborjbernat@gmail.com>
Sponsor: TBD

> Discussions-To: https://discuss.python.org/t/draft-pep-editable-installs-for-pep-517-style-build-backends/8510

Reviewer: Use the topic about this specific PEP draft, rather than PEP 660's topic?

```suggestion
Discussions-To: https://discuss.python.org/t/9071
```

The frontend must create a ``direct_url.json`` file in the ``.dist-info``
directory of the installed distribution, in compliance with PEP 610. The ``url``
value must be a ``file://`` URL pointing to the project directory (i.e., the
directory containing ``pyproject.toml``), and the ``dir_info`` value must be
``{'editable': true}``.

The frontend must not rely on the ``prepare_metadata_for_build_wheel`` hook when
installing in editable mode. It must instead invoke ``build_editable`` and use
the ``.dist-info`` folder returned by that.

Example editable implementations
--------------------------------

To show how this PEP might be used, we'll now present a few case studies. Note
the offered solutions are purely for illustrating purpose.

Add the source tree as is to the interpreter
''''''''''''''''''''''''''''''''''''''''''''

This is one of the simplest implementations, it will add the source tree as is
into the interpreters schema paths.

.. code::

   {
     "metadata_for_build_editable": "<dir to dist-info>",
     "purelib": {"<project dir>": "<project dir>"}
   }

The frontend then could either:

-  Add the source directory onto the target interpreters ``sys.path`` during

>    startup of it. This is done by creating a ``pth`` file into the target

Reviewer 2: ```suggestion
   startup of the interpreter. This is done by creating a ``pth`` file into the target
```

   interpreters ``purelib`` folder. setuptools_ does this today and is what `pip

>    install -e <project_directory>`_ translate too. This solution is fast and

Reviewer 2: ```suggestion
   install -e <project_directory>`_ translates to. This solution is fast and
```
